[Feat] 네비게이션 컨트롤러 구현

cchanmi opened the pull request:
## 🥕 관련 이슈

- Relative : #3 

## 🥕 구현/변경 사항 및 이유

✨ 충돌 나고 멘붕 와서 그냥 다시 했어요 .. ... 앞으로 PR 날리기 전에 꼭 pull 하기 뼈에 새깁니다
- 네비게이션 바, 버튼 구현
- 네비게이션 홈 버튼 눌렀을 때 root 네비게이션 컨트롤러로 이동하는 코드 구현

## 🥕 PR Point

- 제가 만든 폴더는 DetailItem과 asset에서 Navigation 폴더입니다 폴더링 체크 부탁합니다!

## 🥕 참고 사항

Commit:
[Feat] 네비게이션 컨트롤러 구현

noah0316 commented on Danggn.xcodeproj/project.pbxproj:
```diff
@@ -220,6 +249,7 @@
 		4FE39DAB2833DB55002D99B6 /* Screen */ = {
 			isa = PBXGroup;
 			children = (
+				4A08289528357A6E00A582CD /* DetailItem */,
```
DetailItem폴더안에 들어가있는 scene이 Home에 포함되는 scene이니까 Home폴더 안으로 들어가는 건 어떨까요?
> ex) Home/Controller/DetailItemViewController
> Home/View/DetailItem.storyboard

추가로 DetailItem보다 ItemDetail이 소문자 l이랑 대문자I가 겹쳐있지 않아서 가독성이 더 좋은 것 같은데 어떻게 생각하시는지 궁금합니다!

cchanmi replied:
아 저도 가독성이 조금 떨어진다고 생각하였는데 그렇게 하니 훨씬 보기 좋네요! 수정하겠습니다 😇

noah0316 commented on Danggn/Screen/DetailItem/Controller/DetailItemViewController.swift:
```diff
@@ -0,0 +1,20 @@
+
+    }
+
+    @IBAction func homeButton(_ sender: Any) {
```
이 메소드를 호출하는게 UIButton이니까 sender를 UIButton으로 명시적으로 표시해주는게 어떨까요~?🧐
```suggestion
    @IBAction func homeButton(_ sender: UIButton) {
```

cchanmi replied:
수정하였습니다 ~

noah0316 replied:
우리 회의 때 정한 컨벤션대로 메소드 명을  `homeButtonDidTap`으로 가는 것도 좋을 것 같아요!
```suggestion
    @IBAction func homeButtonDidTap(_ sender: UIButton) {
```

cchanmi replied:
엇 이거를 놓쳤네요 지금 수정하겠습니다

cchanmi replied:
이걸 늦게 봐서 ㅠ_ㅠ ... 다시 커밋 날렸습니다

noah0316 reviewed (changes requested):
고생하셨어요~!!👍👍
PR보낸 후에 remote default branch(develop)와 conflict가 난다면 작업중인 브랜치에서 remote default branch를 pull 받은 다음 conflict를 해결하신 후에 다시 push하시면 conflict를 해결할 수 있습니다!!
추가로 GitHub Web Editor를 이용하는 방법도 있습니다!

Commit:
[Chore] 코드 리뷰 수정

noah0316 commented on Danggn/Screen/DetailItem/Controller/DetailItemViewController.swift:
```diff
@@ -0,0 +1,20 @@
+//
```
DetailItem 폴더를 삭제하실 때 Remove Refereces를 선택하신 것 같습니다.
프로젝트 파일에서는 지워져있지만, 실제 파일트리에는 반영이 안되어있네요! 사용하지 않는 파일 삭제 부탁드립니다!
<img width="284" alt="Screen Shot 2022-05-19 at 16 22 54" src="https://user-images.githubusercontent.com/63908856/169235647-b22f7df7-a21a-4d76-a0e8-9cd05f43c6e3.png">
<img width="363" alt="Screen Shot 2022-05-19 at 16 24 15" src="https://user-images.githubusercontent.com/63908856/169235939-5e08e7aa-0b07-4ebb-99ec-7dab6c415d22.png">

cchanmi replied:
어라라,,, 확인하고 삭제하였습니다
그런데 ItemDetail 파일이 xCode에서는 제대로 폴더링이 되어 있고, 실제 트리에서는 ItemDetail이 보이지 않는데 이것은 괜찮은 걸까요...?

noah0316 replied:
@cchanmi 
말씀하시는 파일트리가 프로젝트 내에서의 파일트리를 말씀하시는걸까요?
혹시 터미널이나 파인더를 통해 파일구조를 확인해보셨을까요?!

cchanmi replied:
파인더를 통해 파일구조를 확인하였고
삭제되지 않은 폴더를 삭제하였는데, xCode로는 수정한 파일이 있는데 파인더를 통해서는 수정한 파일이 보이지 않아서요!

- xCode
![스크린샷 2022-05-19 오후 5 58 28](https://user-images.githubusercontent.com/86944161/169255125-14d1075a-28ef-4dec-ae38-1f136219e010.png)

- 파인더
![스크린샷 2022-05-19 오후 5 59 38](https://user-images.githubusercontent.com/86944161/169255391-c67a84f6-2958-4f02-afaa-467f39ca0a7a.png)

noah0316 replied:
Home 폴더 내부 파일구조를 보여주실 수 있을까요?

cchanmi replied:
아 그렇네요,,, 제가 바보였네요,, 죄송합니다

noah0316 replied:
어우 아닙니다 ㅎㅎ GitHub PR에 올라온 내용에 파일(DetailItem 파일)이 삭제되지 않고 그대로 있어서 말씀 드린 것이었습니다 ㅎㅎ
저도 단어를 애매모호하게 말한 것 같아 죄송합니다 ㅎㅎ
삭제하신 후에 commit하시고 다시 push해주세요!

noah0316 replied:
이 폴더와 안에 있는 내용들 삭제 후 PR에 반영해주세요!!!

cchanmi replied:
@noah0316 다시 올렸습니다
제가 계속 똑같은 패턴을 반복하고 있는 것 같아서 무엇이 문제인지 많이 고민을 해 보았는데
승현님이 말씀해 주신 파일들을 삭제하고, stage에 생긴 삭제 기록을 커밋하여야 하는데
이것저것 만지면서 stage 기록을 삭제해 버려서 이런 현상이 반복되었던 것 같네요 ...
이런 패턴이 반복되면서 저 스스로는 삭제를 했다고 생각을 하였는데 PR에는 그것이 반영되지 않았던 것 같습니다 ㅠ___ㅠ
너무 별거 아닌 부분에서 자꾸 시간을 끌어서 너무 죄송하고,,, 정성스럽게 도와주셔서 너무 감사합니다

noah0316 commented on Danggn/Screen/Home/View/ItemDetail.storyboard:
```diff
@@ -0,0 +1,71 @@
+        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
+    </dependencies>
+    <scenes>
+        <!--Navigation Controller-->
```
@cchanmi 
ListItemViewController를 rootViewController로 되어있는 navigationStack에 ItemDetailViewController가
담길예정이니 이친구는 Navigation Controller를 embed하고 있을 필요가 없습니다!
추가로 ItemDetailViewController가 initial ViewController가 되게 해주세요!

이해를 돕기 위한 Navigation Stack 그림입니다!
<img src="https://user-images.githubusercontent.com/63908856/169238577-04a91e62-e6cb-485d-b0ff-32bf66616941.jpeg" width="300" />


이렇게 동작하기 위해서는 담인님 Home Tab의 시작점이 되는 ListViewController가 navigation controller를 embed하고 있어야합니다!
@daminoworld 두분 ListItemViewController와 ItemDetailViewController 네이밍 맞추는 김에 embed 해주시면 좋을 것 같아요!

<img width="541" alt="Screen Shot 2022-05-19 at 16 41 27" src="https://user-images.githubusercontent.com/63908856/169239663-69187f9a-8fb2-4cfd-b864-0e2b64ca55b9.png">

제가 담인님 PR 어제 리뷰드릴 때 이 부분을 고려했어야했는데 여기까지는 미처 생각못했네요 죄송합니다🥲

설정 완료해주시면 아래와 같이 동작할 겁니다!

| iPhone 13 mini 	|
|----------------	|
|![Simulator Screen Recording - iPhone 13 mini - 2022-05-19 at 16 31 45](https://user-images.githubusercontent.com/63908856/169237546-bee19bb4-e5a9-46f6-89d4-f0f85376bd81.gif)|

cchanmi replied:
인스펙터창에서 top bar를 통해 네비게이션 상단바를 보이게 하는 것을 처음 알았네요 !!
그림까지 그려 주시며 너무 자세하게 설명해 주셔서 감사합니다 ㅠㅠ
과제 때 네비게이션 컨트롤러를 사용을 안 했어서 이번에 다루는데에 많이 미숙했는데 덕분에 어떤 것들이 문제였는지 무엇을 더 공부하면 좋을지 알게 되었어요 ✨

cchanmi replied:
정말정말 고생하셨습니다 홍센세 🙏

noah0316 replied:
아이고 감사합니다~~!!!! ㅎㅎㅎㅎㅎ

noah0316 commented on Danggn/Screen/Home/View/ItemDetail.storyboard:
```diff
@@ -0,0 +1,71 @@
+                        <color key="backgroundColor" systemColor="systemBackgroundColor"/>
+                    </view>
+                    <navigationItem key="navigationItem" id="4tw-wQ-ywQ">
+                        <barButtonItem key="leftBarButtonItem" style="plain" id="cZy-FU-IXP">
+                            <button key="customView" opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="center" contentVerticalAlignment="center" buttonType="system" lineBreakMode="middleTruncation" id="yuJ-xR-HIn">
+                                <rect key="frame" x="20" y="3" width="48" height="38"/>
+                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
+                                <state key="normal" title="Button"/>
+                                <buttonConfiguration key="configuration" style="plain" image="ios_icon_home"/>
+                            </button>
+                            <connections>
+                                <action selector="homeButton:" destination="Y6W-OH-hqX" id="cr6-cg-Qqy"/>
+                            </connections>
+                        </barButtonItem>
```
현재 action이 barButtonItem에 연결되어있네요! 이러면 버튼을 터치해도 homeButton 메소드가 원하는대로 실행되지 않을겁니다!
button에 action을 연결해주세요!
```suggestion
                        <barButtonItem key="leftBarButtonItem" style="plain" id="cZy-FU-IXP">
                            <button key="customView" opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="center" contentVerticalAlignment="center" buttonType="system" lineBreakMode="middleTruncation" id="yuJ-xR-HIn">
                                <rect key="frame" x="20" y="3" width="48" height="38"/>
                                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                                <state key="normal" title="Button"/>
                                <buttonConfiguration key="configuration" style="plain" image="ios_icon_home"/>
                                <connections>
                                    <action selector="homeButton:" destination="Y6W-OH-hqX" eventType="touchUpInside" id="LcL-yR-6ir"/>
                                </connections>
                            </button>
                        </barButtonItem>
```
<img width="840" alt="Screen Shot 2022-05-19 at 16 50 51" src="https://user-images.githubusercontent.com/63908856/169241488-b0ecf8ef-670a-45e1-b6da-56974f51988e.png">

cchanmi replied:
아 저는 연결된 기능이 없어서 눌리지가 않는 줄 알았는데 그것이 아니었군요...!
놓친 게 많네요 ㅠㅠ 수정하겠습니다

Commit:
[Chore] 코드 리뷰 수정

- 버전은 제가 잘못 건드려서 다시 원상복구 해 놓은 겁니다,,
- Button으로 Action 함수 다시 연결
- navigation 연결 수정

Commit:
[Chore] 홈 버튼명 수정

Commit:
[Chore] homeButton 액션 제거

Commit:
[Chore] DetailItem 파일 삭제

Commit:
[Chore] 빈 에셋 삭제, 에셋 이름 수정

ex) ios_icon_home > icon_home으로 수정

Commit:
[Chore] 빈 에셋 삭제, 에셋 이름 수정

Commit:
Merge remote-tracking branch 'origin/feature/2-navigation-bar' into feature/2-navigation-bar

Commit:
[Chore] 네비게이션 상단바 버튼 이미지 다시 설정

에셋 이름 바꾼 후, 이미지 다시 설정

Commit:
[Chore] barButtonItem 기본값으로 복구

noah0316 reviewed (approved):
고생하셨어요~!! 최고!!ㅎㅎㅎ👍
